### ISSUE TITLE
加载成功 但是无法使用

### ISSUE THREAD

**Issue by LaziestDog** [chr233/ASFEnhance#206]
**Describe the bug / Bug 描述**
加载成功 但是IPC内 `命令` 输入所有命令都是提示未知命令

**Error Log / 错误日志**

信息
```txt
 - Access: Owner
 - ASF 版本: ASF 5.5.1.4 - linux-x64 - d9c38ab
 - 插件版本: ASFEnhance 版本 2.0.10.0
==========================================
{
  "EULA": true,
  "Statistic": true,
  "DevFeature": true
}
==========================================
日志内无其他相关内容
> ASFE
< <Little> 未知命令！
> CLAIMITEM
< <Little> 未知命令！
```
加载成功信息
```txt
 > ==========================================================================
> █████╗ ███████╗███████╗███████╗███╗ ██╗██╗ ██╗ █████╗ ███╗ ██╗ ██████╗███████╗
> ██╔══██╗██╔════╝██╔════╝██╔════╝████╗ ██║██║ ██║██╔══██╗████╗ ██║██╔════╝██╔════╝
> ███████║███████╗█████╗ █████╗ ██╔██╗ ██║███████║███████║██╔██╗ ██║██║ █████╗
> ██╔══██║╚════██║██╔══╝ ██╔══╝ ██║╚██╗██║██╔══██║██╔══██║██║╚██╗██║██║ ██╔══╝
> ██║ ██║███████║██║ ███████╗██║ ╚████║██║ ██║██║ ██║██║ ╚████║╚██████╗███████╗
> ╚═╝ ╚═╝╚══════╝╚═╝ ╚══════╝╚═╝ ╚═══╝╚═╝ ╚═╝╚═╝ ╚═╝╚═╝ ╚═══╝ ╚═════╝╚══════╝
> ==========================================================================
> ASFEnhance 版本 2.0.10.0
> 插件作者: Chr_, 联系方式: chr@chrxw.com
> 爱发电: https://afdian.net/@chr233
> ==========================================================================
> 未加载外部模块
> ==========================================================================
> 使用 "PLUGINVERSION [插件名]" / "PV [插件名]" 检查插件可用更新
> 使用 "PLUGINUPDATE [插件名]" / "PU [插件名]" 自动更新指定插件
> 不指定插件名时, 可为所有支持的插件检查更新 / 安装更新
> ==========================================================================
>
[04:36:16] WARN ASF > OnASFInit()
> ==========================================================================
> ASFEhance 开发者特性已开启
> ==========================================================================
```

**Comment by chr233** [comment]
还有更多日志吗，ASF的日志有吗

**Comment by LaziestDog** [comment]
抱歉，asf日志并没有更多相关内容，有的我已经全部贴上去了，现在家里服务器异常终止，我这几天也没办法提供更多信息

Chr_ ***@***.***> 于 2024年1月26日周五 09:53写道：

**Comment by psy0ch** [comment]
Response from bot:

`ASFenhance encountered error, error logs below
==========================================================================
- Origin Message: pluginlist
- Access: Owner
- ASF Version: 5.5.2.3
- Plugin Version: 2.0.10.1
==========================================================================
{
"EULA": true,
"Statistic": false,
"DevFeature": false,
"DisabledCmds": null,
"Address": null,
"Addresses": [],
"ApiKey": null,
"AutoClaimItemBotNames": "x",
"AutoClaimItemPeriod": 23
}
==========================================================================
- Exception Type: System.InvalidCastException
- Exception Message: Unable to cast object of type 'System.Collections.Frozen.DefaultFrozenSet`1[ArchiSteamFarm.Plugins.Interfaces.IPlugin]' to type 'System.Collections.Immutable.ImmutableHashSet`1[ArchiSteamFarm.Plugins.Interfaces.IPlugin]'.
at ASFEnhance.Explorer.ReflectionHelper.GetStaticPrivateProperty[T](Type type, String fieldname)
at ASFEnhance.Update.Command.ResponsePluginList()
at ASFEnhance.ASFEnhance.ResponseCommand(Bot bot, EAccess access, String cmd, String message, String[] args, UInt64 steamId)
at ASFEnhance.ASFEnhance.OnBotCommand(Bot bot, EAccess access, String message, String[] args, UInt64 steamId)`

**Comment by chr233** [comment]
已更新，使用2.0.11.0

**Comment by Gensokyo1995** [comment]
> 已更新，使用2.0.11.0

测试了下大部分指令都正常
但是PLUGINSLIST和PLUGINLIST指令仍然报错

ASFenhance 遇到错误, 日志如下
==========================================================================
- 原始消息: PLUGINSLIST
- Access: Owner
- ASF 版本: 5.5.2.3
- 插件版本: 2.0.11.0
==========================================================================
{
"EULA": true,
"Statistic": true,
"DevFeature": true,
"DisabledCmds": [],
"Address": null,
"Addresses": [],
"ApiKey": null,
"AutoClaimItemBotNames": "123",
"AutoClaimItemPeriod": 23
}
==========================================================================
- 错误类型: System.InvalidCastException
- 错误消息: Specified cast is not valid.
at ASFEnhance.Explorer.ReflectionHelper.GetStaticPrivateProperty[T](Type type, String fieldname)
at ASFEnhance.Update.Command.ResponsePluginList()
at ASFEnhance.ASFEnhance.ResponseCommand(Bot bot, EAccess access, String cmd, String message, String[] args, UInt64 steamId)
at ASFEnhance.ASFEnhance.OnBotCommand(Bot bot, EAccess access, String message, String[] args, UInt64 steamId)

**Comment by psy0ch** [comment]
Different Error for me, same versions

`ASFenhance encountered error, error logs below
==========================================================================
- Origin Message: pluginlist
- Access: Owner
- ASF Version: 5.5.2.3
- Plugin Version: 2.0.11.0
==========================================================================
{
"EULA": true,
"Statistic": false,
"DevFeature": false,
"DisabledCmds": null,
"Address": null,
"Addresses": [],
"ApiKey": null,
"AutoClaimItemBotNames": "ASF",
"AutoClaimItemPeriod": 23
}
==========================================================================
- Exception Type: System.InvalidCastException
- Exception Message: Unable to cast object of type 'System.Collections.Frozen.DefaultFrozenSet`1[ArchiSteamFarm.Plugins.Interfaces.IPlugin]' to type 'System.Collections.Immutable.ImmutableHashSet`1[ArchiSteamFarm.Plugins.Interfaces.IPlugin]'.
at ASFEnhance.Explorer.ReflectionHelper.GetStaticPrivateProperty[T](Type type, String fieldname)
at ASFEnhance.Update.Command.ResponsePluginList()
at ASFEnhance.ASFEnhance.ResponseCommand(Bot bot, EAccess access, String cmd, String message, String[] args, UInt64 steamId)
at ASFEnhance.ASFEnhance.OnBotCommand(Bot bot, EAccess access, String message, String[] args, UInt64 steamId)`

**Comment by chr233** [comment]
> Different Error for me, same versions
> 
> `ASFenhance encountered error, error logs below
> ==========================================================================
> - Origin Message: pluginlist
> - Access: Owner
> - ASF Version: 5.5.2.3
> - Plugin Version: 2.0.11.0
> ==========================================================================
> {
> "EULA": true,
> "Statistic": false,
> "DevFeature": false,
> "DisabledCmds": null,
> "Address": null,
> "Addresses": [],
> "ApiKey": null,
> "AutoClaimItemBotNames": "ASF",
> "AutoClaimItemPeriod": 23
> }
> ==========================================================================
> - Exception Type: System.InvalidCastException
> - Exception Message: Unable to cast object of type 'System.Collections.Frozen.DefaultFrozenSet`1[ArchiSteamFarm.Plugins.Interfaces.IPlugin]' to type 'System.Collections.Immutable.ImmutableHashSet`1[ArchiSteamFarm.Plugins.Interfaces.IPlugin]'.
> at ASFEnhance.Explorer.ReflectionHelper.GetStaticPrivateProperty[T](Type type, String fieldname)
> at ASFEnhance.Update.Command.ResponsePluginList()
> at ASFEnhance.ASFEnhance.ResponseCommand(Bot bot, EAccess access, String cmd, String message, String[] args, UInt64 steamId)
> at ASFEnhance.ASFEnhance.OnBotCommand(Bot bot, EAccess access, String message, String[] args, UInt64 steamId)`

got it

**Comment by chr233** [comment]
> > 已更新，使用2.0.11.0
> 
> 测试了下大部分指令都正常
> 但是PLUGINSLIST和PLUGINLIST指令仍然报错(应该就这两个有报错了)
> 
> ASFenhance 遇到错误, 日志如下
> ==========================================================================
> - 原始消息: PLUGINSLIST
> - Access: Owner
> - ASF 版本: 5.5.2.3
> - 插件版本: 2.0.11.0
> ==========================================================================
> {
> "EULA": true,
> "Statistic": true,
> "DevFeature": true,
> "DisabledCmds": [],
> "Address": null,
> "Addresses": [],
> "ApiKey": null,
> "AutoClaimItemBotNames": "123",
> "AutoClaimItemPeriod": 23
> }
> ==========================================================================
> - 错误类型: System.InvalidCastException
> - 错误消息: Specified cast is not valid.
> at ASFEnhance.Explorer.ReflectionHelper.GetStaticPrivateProperty[T](Type type, String fieldname)
> at ASFEnhance.Update.Command.ResponsePluginList()
> at ASFEnhance.ASFEnhance.ResponseCommand(Bot bot, EAccess access, String cmd, String message, String[] args, UInt64 steamId)
> at ASFEnhance.ASFEnhance.OnBotCommand(Bot bot, EAccess access, String message, String[] args, UInt64 steamId)
> 

了解了，明天修

**Comment by chr233** [comment]
fixed in 2.0.11.1
